File: src/main/scala/com/yogpc/qp/block/DummyBlock.scala
package com.yogpc.qp.block

import com.yogpc.qp.{QuarryPlus, QuarryPlusI}
import net.minecraft.block.BlockEmptyDrops
import net.minecraft.block.material.Material
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.entity.{Entity, EntityLiving}
import net.minecraft.item.ItemBlock
import net.minecraft.util.math.BlockPos
import net.minecraft.util.{BlockRenderLayer, EnumFacing}
import net.minecraft.world.{IBlockAccess, World}
import net.minecraftforge.fml.relauncher.{Side, SideOnly}

class DummyBlock extends BlockEmptyDrops(Material.GLASS) {
    setRegistryName(QuarryPlus.modID, QuarryPlus.Names.dummyblock)
    setUnlocalizedName(QuarryPlus.modID + "." + QuarryPlus.Names.dummyblock)
    setCreativeTab(QuarryPlusI.creativeTab)
    setHardness(1.0f)
    setLightOpacity(0)
    setLightLevel(1f)
    disableStats()

    val itemBlock = new ItemBlock(this)
    itemBlock.setRegistryName(QuarryPlus.modID, QuarryPlus.Names.dummyblock)

    override def canCreatureSpawn(state: IBlockState, world: IBlockAccess, pos: BlockPos, t: EntityLiving.SpawnPlacementType): Boolean = false

    override def canEntitySpawn(state: IBlockState, entityIn: Entity): Boolean = false

    override def canSilkHarvest(world: World, pos: BlockPos, state: IBlockState, player: EntityPlayer): Boolean = false

    @SideOnly(Side.CLIENT)
    override def getBlockLayer = BlockRenderLayer.CUTOUT

    override def isFullCube(state: IBlockState) = false

    override def isOpaqueCube(state: IBlockState) = false

    //noinspection ScalaDeprecation
    @SideOnly(Side.CLIENT)
    @SuppressWarnings("deprecation")
    override def shouldSideBeRendered(blockState: IBlockState, blockAccess: IBlockAccess, pos: BlockPos, side: EnumFacing): Boolean = {
        val offsetState = blockAccess.getBlockState(pos offset side)
        offsetState.getBlock != this && super.shouldSideBeRendered(blockState, blockAccess, pos, side)
    }

}

File: src/main/scala/com/yogpc/qp/block/DummyBlock.java
package com.yogpc.qp.block;

import com.yogpc.qp.QuarryPlus;
import com.yogpc.qp.QuarryPlusI;
import net.minecraft.block.BlockEmptyDrops;
import net.minecraft.block.material.Material;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityLiving;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemBlock;
import net.minecraft.util.BlockRenderLayer;
import net.minecraft.util.EnumFacing;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockAccess;
import net.minecraft.world.World;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public class DummyBlock extends BlockEmptyDrops {
    private final ItemBlock mItemBlock;

    public DummyBlock() {
        super(Material.GLASS);
        setRegistryName(QuarryPlus.modID, QuarryPlus.Names.dummyblock);
        setUnlocalizedName(QuarryPlus.modID + "." + QuarryPlus.Names.dummyblock);
        setCreativeTab(QuarryPlusI.creativeTab());
        setHardness(1.0f);
        setLightOpacity(0);
        setLightLevel(1f);
        disableStats();
        mItemBlock = new ItemBlock(this);
        mItemBlock.setRegistryName(QuarryPlus.modID, QuarryPlus.Names.dummyblock);
    }

    public ItemBlock itemBlock() {
        return mItemBlock;
    }

    @Override
    public boolean canCreatureSpawn(IBlockState state, IBlockAccess world, BlockPos pos, EntityLiving.SpawnPlacementType type) {
        return false;
    }

    @Override
    @SuppressWarnings("deprecation")
    public boolean canEntitySpawn(IBlockState state, Entity entityIn) {
        return false;
    }

    @Override
    public boolean canSilkHarvest(World world, BlockPos pos, IBlockState state, EntityPlayer player) {
        return false;
    }

    @SideOnly(Side.CLIENT)
    @Override
    public BlockRenderLayer getBlockLayer() {
        return BlockRenderLayer.CUTOUT;
    }

    @Override
    @SuppressWarnings("deprecation")
    public boolean isFullCube(IBlockState state) {
        return false;
    }

    @Override
    @SuppressWarnings("deprecation")
    public boolean isOpaqueCube(IBlockState state) {
        return false;
    }

    @SideOnly(Side.CLIENT)
    @Override
    @SuppressWarnings("deprecation")
    public boolean shouldSideBeRendered(IBlockState blockState, IBlockAccess blockAccess, BlockPos pos, EnumFacing side) {
        IBlockState offsetState = blockAccess.getBlockState(pos.offset(side));
        return offsetState.getBlock() != this && super.shouldSideBeRendered(blockState, blockAccess, pos, side);
    }

}
